src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs

```diff
@@ -62,8 +60,17 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                 // This method is the strongly-typed Equals method where the parameter type is
                 // the containing type.
 
-                if (ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType())
+                bool isRecordStruct = ContainingType.IsStructType();
+                if (isRecordStruct)
                 {
+                    // We'll produce:
+                    // virtual bool Equals(T other) =>
+                    //     field1 == other.field1 && ... && fieldN == other.fieldN;
+                    retExpr = F.Literal(true);
```

Comment:
>F.Literal(true); [](start = 30, length = 16)

This doesn't look usefull. I think we can use null value instead and avoid generating unnecessary nodes here and below.

Reply:
We need a true one way or another for the case with no fields. I'll move it to only that case

---
In reply to: [597075436](https://github.com/dotnet/roslyn/pull/51900#discussion_r597075436) [](ancestors = 597075436)